src/gc/boehm.cr

```diff
@@ -254,20 +254,22 @@ module GC
   end
 
   # pushes the stack of pending fibers when the GC wants to collect memory:
-  GC.before_collect do
-    Fiber.unsafe_each do |fiber|
-      fiber.push_gc_roots unless fiber.running?
-    end
+  {% unless flag?(:win32) %}
```

Comment:
I don't understand the reason for this change.
The GC in windows still needs to push the gc root, and the extensions are already behind preview_mt.

Reply:
Master fails to compiler because `Fiber` is not yet implemented on win32. It is only added in #7995

Reply:
This is only temporary and will be removed in #7995. But it's necessary to run the specs prior to that.